File: Zig/fibonacci.zig
const out = @import("std").io.getStdOut().writer();

pub fn main() !void {
    var a: u64 = 0;
    try out.print("{}\n", .{a});
    var b: u64 = 1;
    try out.print("{}\n", .{b});

    var c: u64 = a +% b;
    while (c >= b) : (c = a +% b) {
        try out.print("{}\n", .{c});
        a = b;
        b = c;
    }
    try out.print("Overflowed!\n", .{});
    return;
}

File: Rust/src/fibonacci.rs
// Legacy fibonacci function which prints all within u64 bounds
fn main() {
    let mut a: u64 = 0;
    println!("{}", a);
    let mut b: u64 = 1;
    println!("{}", b);

    while let Some(c) = a.checked_add(b) {
        println!("{}", c);
        a = b;
        b = c;
    }
    println!("Overflowed!");
}
